projects/github-actions/required-review/src/requirement.js

```diff
@@ -31,10 +32,12 @@ function buildReviewerFilter( config, teamConfig, indent ) {
 		const team = teamConfig;
 		return async function ( reviewers ) {
 			const members = await fetchTeamMembers( team );
-			return printSet(
-				`${ indent }Members of ${ team }:`,
-				reviewers.filter( reviewer => members.includes( reviewer ) )
-			);
+			const reviewSatisfied = reviewers.filter( reviewer => members.includes( reviewer ) );
+			const request = core.getInput('request-reviews')
+			if ( reviewSatisfied.length === 0 && request == 'true' ) {
+				await requestReview( team );
+			}
+			return printSet( `${ indent }Members of ${ team }:`, reviewSatisfied );
```

Comment:
Thanks for the review @anomiex  After going through the code I went this route due to how reviewers seemed to be grabbed first and then it looked like it worked back to the teams. This route seemed the most direct w/o making major changes to the rest of the code. With that said, I agree 💯 with limiting unneeded API calls and will look into reducing these as you're suggesting.

I'm wondering if it might be reasonable to say that review requests only occur on the first run? On that initial run I think it's safe to say that there would be no reviews therefor all required reviewers should be requested. Then as reviews are received, there shouldn't be the further need to request them again, as you point out. There could be a case where a review was removed (new commit push, etc), but thinking of how codeowners files works, it only requests reviews the first time iirc, which is the functionality I'm looking to introduce. Not sure if this would make the above any more straight forward.

Reply:
If someone pushes a commit to a PR that touches additional files, that might also trigger a need for more reviewers. 

For example, if someone from teamX says "oh, you also need to change this other thing over here" in their review, and that other thing happens to require review from teamZ.

Reply:
yup - great point. 👍🏻